server/server.go

```diff
@@ -220,7 +232,7 @@ func configureBlockchain(conf *Config, store storage.Store) (*emulator.Blockchai
 	if conf.ServicePublicKey != nil {
 		options = append(
 			options,
-			emulator.WithServicePublicKey(conf.ServicePublicKey, conf.ServiceKeySigAlgo, conf.ServiceKeyHashAlgo),
+			emulator.WithServiceKey(conf.ServicePublicKey, conf.ServicePrivateKey, conf.ServiceKeySigAlgo, conf.ServiceKeyHashAlgo),
```

Comment:
what if they only pass public key and not private key but use the option to deploy advanced contracts. This edge case should be handled.